### ISSUE TITLE
Compilation Errors in src/ivf_rabitq.h Related to Memory Release Operations on Ubuntu 22.04 with GCC 11.4

### ISSUE THREAD

**Issue by GatePara** [gaoj0017/RaBitQ#1]
### Issue Description
I encountered several compilation errors in the ./src/index.cpp file, related to memory release operations in src/ivf_rabitq.h. The error messages are as follows:
![120891717078611_ pic](https://github.com/gaoj0017/RaBitQ/assets/73482395/2ac33d09-b0d2-4375-82cd-bebfa8a0a7f3)

### Steps to Reproduce

1. Clone the repository
2. Prepare sift dataset
3. Execute the build script: ./script/index.sh
4. Observe the errors

### Attempted Solutions
I tried commenting out the line 464 in src/ivf_rabitq.h, specifically std::free(pack_codes);. This allows the compilation to pass, but it might lead to potential memory leaks.

### Assistance Requested
Is there a more appropriate way to handle this memory release?
Is this issue due to the compiler configuration or a problem with the code itself?

### Environment
Operating System: Ubuntu 22.04 LTS
Compiler Version: GCC 11.4

**Comment by gaoj0017** [comment]
Thanks for letting me know the bug. Yes. It is a problem of the code itself. I realized the issue of memory leakage just before uploading this repo. I tried to resolve this with ``std::free'' but then I uploaded this repo without testing. It turns out that it does not work :(

Let me first comment out this line such that it can at least pass compilation and handle ANN queries normally. I shall fix the issue of memory release later on.
